docs/concepts.md

```diff
@@ -0,0 +1,123 @@
+        else:
+            data["correct_answers"]["y"] = 3 * data["params"]["x"]
+
+    def parse(data):
+        if "y" not in data["format_errors"] and data["submitted_answers"]["y"] < 0:
+            data["format_errors"]["y"] = "Negative numbers are not allowed"
+
+    def grade(data):
+        if math.isclose(data["score"], 0.0) and data["submitted_answers"]["y"] > data["params"]["x"]:
+            data["partial_scores"]["y"]["score"] = 0.5
+            data["score"] = 0.5
+            data["feedback"]["y"] = "Your value for $y$ is larger than $x$, but incorrect."
+    ```
+
+=== "Example `info.json`"
+
+    ```json title="info.json"
+    {
+        "uuid": "cbf5cbf2-6458-4f13-a418-aa4d2b1093ff",
+        "title": "Doubling and Tripling",
+        "topic": "Basic Arithmetic",
+        "tags": ["sp24"],
+        "type": "v3",
+    }
+    ```
+
+### Elements
+
+Questions are written in HTML and **Elements**. Elements are pre-built interface elements for common questions, like numerical inputs, multiple choice, and more. **Custom Elements** are reusable components that can be tailored to your course and provide a way to create more complex questions. Custom elements are not necessary for most courses.
+
+=== "Integer input element"
+
+    ![](elements/pl-integer-input.png)
+
+=== "HTML for element"
+
+    ```html title="question.html"
+    <pl-integer-input answers-name="int_value" label="$y =$"></pl-integer-input>
+    ```
+
+You can view a list of all the available elements in the [Elements documentation](./elements.md).
+
+### Assessments
+
+A course contains a collection of questions, which are composed together to create [**Assessments**](./assessment/index.md). Each course instance contains a collection of assessments. There are **Exam Assessments** and **Homework Assessments**.
+
+Asssessments are organized into **Assessment Sets** based on the type of assessment (e.g. `Homework`, `Quiz`, `Exam`). Optionally, each assessment can be a part of a **Module** (e.g. `Introduction`, `Review`, `Linked Lists`) that represent a segment of a course. Your assessments can then be shown to students grouped by module or by assessment set.
+
+=== "Assessments (Module Grouping)"
+
+    ![alt text](img/module-grouping.png)
+
+=== "Assessments (Set Grouping)"
+
+    ![alt text](img/set-grouping.png)
+
+### Metadata
+
+Each item in PrairieLearn (questions, assessments, etc.) has associated metadata that describes the item. This metadata is stored in JSON files and describes the relationships between items, the item's properties, and other information. The metadata files are used to generate the user interface for editing and viewing the items, as well as the interface for students to complete the items.
+
+| Item            | Metadata file         |
+| --------------- | --------------------- |
+| Question        | `info.json`           |
+| Course          | `infoCourse.json`     |
+| Course Instance | `infoAssessment.json` |
```

Comment:
This need to be split into "course instance" and "assessment" and reference the appropriate file for each.